### PR TITLE
Exec-Command should re-use console

### PR DESCRIPTION
**Pull request by jaredpar**
Change the Exec-Command to re-use the current console vs. redirecting all
of the output. This lets items like color, and cancel work more naturally for
spawned processes.

### CODE REVIEW

**Comment by tmeschter** [comment]
@jaredpar Any downsides to this approach?

**Comment by jaredpar** [comment]
@tmeschter actually yeah there is a downside I need to handle. The problem here is that now `Exec-Command` will unconditionally write to the console. Hence any use of this pattern is now broken

``` ps1
Exec-Command $cmd | Out-Null
```

Need to work through this. Shouldn't be too hard.

**Comment by jaredpar** [comment]
@tmeschter okay now we have two functions:

- Exec-Command: same as before. Spawns a process and returns standard out for use in the calling powershell code. Useful for ignoring output or parsing it. 
- Exec-Console: like Exec-Command but gives process the same console so that colors come through.

File: build/scripts/build.ps1 [hunk 1]
```diff
@@ -94,7 +94,7 @@ function Run-MSBuild([string]$buildArgs = "", [string]$logFile = "") {
     }
 
     $args += " $buildArgs"
-    Exec-Command $msbuild $args
+    Exec-Console $msbuild $args
```
Comment (tmeschter) [inline build/scripts/build.ps1]: Should this remain `Exec-Console`?

Comment (jaredpar) [inline build/scripts/build.ps1]: Believe so yes. But should remove all of the `Out-Host` pipes from calls to Run-MSBuild

**Pull request merged**